### PR TITLE
Dynamax support

### PR DESCRIPTION
**Pull request by Fabio1988**
- Stations
- Battle Level + Pokemon
- Stationed Pokemon

unclear if ever needed:
- webhooks for Poracle

for later:
- Prometheus station support for decoding
- Prometheus dashboard update?!

### CODE REVIEW

File: sql/30_station.up.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+CREATE TABLE `station` (
+ `id`                       VARCHAR(35) NOT NULL,
+ `lat`                      DOUBLE(18,14) NOT NULL,
+ `lon`                      DOUBLE(18,14) NOT NULL,
+ `name`                     VARCHAR(128) NOT NULL,
+ `cell_id`                  BIGINT UNSIGNED NOT NULL,
```
Comment (nakratz) [inline sql/30_station.up.sql:6]: All my Cell ID's are negative numbers on the East Coast of the US so Unsigned for cell_id doesn't work and resulted in 
insert station: Error 1264 (22003): Out of range value for column 'cell_id' at row 1

Comment (jfberry) [reply sql/30_station.up.sql:6]: does it work if you adjust the table by hand?

Comment (nakratz) [reply sql/30_station.up.sql:6]: Yes I manually changed the table to no be unsigned and it is now putting them in the table properly.

Comment (Fabio1988) [reply sql/30_station.up.sql]: thanks, fixed... now it's aligned with the other `cell_id` columns, was a copy-paste error :)

**Comment by Fabio1988** [comment]
Either drop table `station` and decrease `schema_migrations.version` so migration can be applied from scratch, or execute manually, no guarantee :) 

```sql
ALTER TABLE `station`
ADD COLUMN `total_stationed_pokemon`  SMALLINT unsigned DEFAULT NULL,
ADD COLUMN `stationed_pokemon`            TEXT DEFAULT NULL;
```

**Comment by Fabio1988** [comment]
full GMO entry of stations for later reference:
```
stations {
            id "a2266dbb66a94c45969ab32a8c44a2e4.23"
            lat x.xxxx
            lng x.xxxx
            name "XXX XXX XXX"
            start_time_ms 1725850800000
            end_time_ms 1726110000000
            is_bread_battle_available true
            
            battle_details {
                bread_battle_seed -8441647197908165682
                battle_spawn_ms 1725850800000
                battle_window_start_ms 1725940800000
                battle_window_end_ms 1725994800000
                battle_level BreadBattleLevel.BREAD_BATTLE_LEVEL_1:1
                
                battle_pokemon {
                    pokemon_id HoloPokemonId.SKWOVET:819
                    stamina 1700
                    max_stamina 1700
                    move1 HoloPokemonMove.BULLET_SEED_FAST:271
                    move2 HoloPokemonMove.CRUNCH:279
                    individual_attack 15
                    individual_defense 15
                    individual_stamina 15
                    cp_multiplier 0.15000000596046448
                    
                    pokemon_display {
                        display_id 819
                        bread_mode_enum Modifier.BREAD_MODE:1
                    }
                }
                
                reward_pokemon {
                    pokemon_id HoloPokemonId.SKWOVET:819
                    stamina 1700
                    max_stamina 1700
                    move1 HoloPokemonMove.BULLET_SEED_FAST:271
                    move2 HoloPokemonMove.CRUNCH:279
                    individual_attack 15
                    individual_defense 15
                    individual_stamina 15
                    cp_multiplier 0.15000000596046448
                    
                    pokemon_display {
                        display_id 819
                        bread_mode_enum Modifier.BREAD_MODE:1
                    }
                }
            }
        }
```

**Comment by Fabio1988** [comment]
whoever pulling latest commit either drop migration or execute:

```sql
ALTER TABLE `station`
ADD COLUMN `battle_start`             INT UNSIGNED,
ADD COLUMN `battle_end`               INT UNSIGNED;
```

**Comment by Fabio1988** [comment]
yeah for now this should be good. tested it and we process the columns like we want them :)

if gigantamax is ever introduced this might just be a visual change on the bread_mode_modifier in the PokemonDisplay of the battle pokemon. there are no fields in the general station proto which indicates if dynamax or gigantamax, it's just visible in the PokemonDisplay, see [here](https://github.com/UnownHash/Golbat/pull/233#issuecomment-2341592679)

**Pull request merged**